### ISSUE TITLE
 One package not publishing properly (3.0.0-beta.18)

### ISSUE THREAD

**Issue by reggi** [lerna/lerna#1407]
> Closing seems to be an issue with `npm`.

I have a monorepo with one package here's it on NPM [@reggi/pkg.json-parse](https://www.npmjs.com/package/@reggi/pkg.json-parse) it shows "version" at `0.0.2`. However It is publishing at `0.0.21`.

```
$ npm view @reggi/pkg.json-parse version
0.0.2
```

Here's the publish command:

```
> lerna publish --yes --cd-version=patch

lerna info version 3.0.0-beta.18
lerna info versioning independent
lerna info Checking for updated packages...
lerna info Comparing with @reggi/command@0.0.8.

Changes:
 - @reggi/command: 0.0.9 => 0.0.10
 - @reggi/dep-merge: 0.0.8 => 0.0.9
 - @reggi/dep-merge-cli: 0.0.6 => 0.0.7
 - @reggi/gitpkg.upsert: 0.0.5 => 0.0.6
 - @reggi/help: 0.0.9 => 0.0.10
 - @reggi/help.filter-until: 0.0.10 => 0.0.11
 - @reggi/help.parse-argv: 0.0.10 => 0.0.11
 - @reggi/help.set-entire: 0.0.10 => 0.0.11
 - @reggi/help.string-argv: 0.0.10 => 0.0.11
 - @reggi/journey: 1.0.8 => 1.0.9
 - @reggi/journey.coerce-to-array: 0.0.10 => 0.0.11
 - @reggi/journey.coerce-to-plain-object: 0.0.10 => 0.0.11
 - @reggi/journey.fn-free: 0.0.9 => 0.0.10
 - @reggi/journey.fn-reduce: 0.0.9 => 0.0.10
 - @reggi/journey.is-promise: 0.0.10 => 0.0.11
 - @reggi/journey.pass-thru: 0.0.9 => 0.0.10
 - @reggi/pkg: 0.0.6 => 0.0.7
 - @reggi/pkg-cli: 0.0.6 => 0.0.7
 - @reggi/pkg-plugin-babel-6-to-node-4: 0.0.9 => 0.0.10
 - @reggi/pkg-plugin-babel-7-to-node-4: 0.0.9 => 0.0.10
 - @reggi/pkg-plugin-cobalt: 0.0.9 => 0.0.10
 - @reggi/pkg-plugin-jest: 0.0.9 => 0.0.10
 - @reggi/pkg-plugin-name-dir: 0.0.10 => 0.0.11
 - @reggi/pkg-plugin-name-scope: 0.0.10 => 0.0.11
 - @reggi/pkg-plugin-prop: 0.0.10 => 0.0.11
 - @reggi/pkg-plugin-sort: 0.0.10 => 0.0.11
 - @reggi/pkg-plugin-standard: 0.0.9 => 0.0.10
 - @reggi/pkg.file-exists: 0.0.9 => 0.0.10
 - @reggi/pkg.fs: 0.0.10 => 0.0.11
 - @reggi/pkg.is-local-module: 0.0.10 => 0.0.11
 - @reggi/pkg.json-parse: 0.0.20 => 0.0.21
 - @reggi/pkg.pretty-json: 0.0.9 => 0.0.10
 - @reggi/pkg.prop-overwrite: 0.0.9 => 0.0.10
 - @reggi/pkg.read-json: 0.0.8 => 0.0.9
 - @reggi/pkg.throw-error: 0.0.9 => 0.0.10
 - @reggi/pkgprop: 0.0.2 => 0.0.3
 - @reggi/pkgprop-cli: 0.0.1 => 0.0.2
 - reggi-cli: 0.0.1 => 0.0.2
 - replace-cli: 1.0.9 => 1.0.10
 - @reggi/requireable: 0.0.10 => 0.0.11
 - @reggi/requireable-cli: 0.0.13 => 0.0.14
 - results-cli: 0.0.9 => 0.0.10
 - @reggi/subrepo: 0.0.2 => 0.0.3
 - @reggi/subrepo-cli: 0.0.1 => 0.0.2

lerna info auto-confirmed 
lerna info publish Publishing packages to npm...
lerna info published @reggi/command
lerna info published @reggi/journey.coerce-to-array
lerna info published @reggi/help.string-argv
lerna info published @reggi/help.filter-until
lerna info published @reggi/help.set-entire
lerna info published @reggi/journey.coerce-to-plain-object
lerna info published @reggi/pkg-plugin-name-scope
lerna info published @reggi/journey.is-promise
lerna info published @reggi/pkg-plugin-name-dir
lerna info published @reggi/pkg-plugin-prop
lerna info published @reggi/pkg-plugin-sort
lerna info published @reggi/pkg.pretty-json
lerna info published @reggi/pkg.is-local-module
lerna info published @reggi/pkg.fs
lerna info published @reggi/pkg.json-parse
lerna info published @reggi/pkg.prop-overwrite
lerna info published @reggi/subrepo
lerna info published @reggi/pkg.throw-error
lerna info published @reggi/pkg-plugin-babel-6-to-node-4
lerna info published @reggi/journey.pass-thru
lerna info published @reggi/pkg-plugin-babel-7-to-node-4
lerna info published @reggi/help.parse-argv
lerna info published @reggi/journey.fn-free
lerna info published @reggi/pkg-plugin-jest
lerna info published @reggi/pkg-plugin-standard
lerna info published @reggi/pkgprop
lerna info published @reggi/pkg.file-exists
lerna info published replace-cli
lerna info published @reggi/journey.fn-reduce
lerna info published @reggi/pkg-plugin-cobalt
lerna info published @reggi/journey
lerna info published @reggi/dep-merge
lerna info published @reggi/help
lerna info published @reggi/gitpkg.upsert
lerna info published @reggi/pkg.read-json
lerna info published @reggi/requireable
lerna info published @reggi/pkg
lerna info published @reggi/pkgprop-cli
lerna info published @reggi/requireable-cli
lerna info published @reggi/dep-merge-cli
lerna info published results-cli
lerna info published @reggi/subrepo-cli
lerna info published @reggi/pkg-cli
lerna info published reggi-cli
Successfully published:
 - @reggi/command@0.0.10
 - @reggi/dep-merge@0.0.9
 - @reggi/dep-merge-cli@0.0.7
 - @reggi/gitpkg.upsert@0.0.6
 - @reggi/help@0.0.10
 - @reggi/help.filter-until@0.0.11
 - @reggi/help.parse-argv@0.0.11
 - @reggi/help.set-entire@0.0.11
 - @reggi/help.string-argv@0.0.11
 - @reggi/journey@1.0.9
 - @reggi/journey.coerce-to-array@0.0.11
 - @reggi/journey.coerce-to-plain-object@0.0.11
 - @reggi/journey.fn-free@0.0.10
 - @reggi/journey.fn-reduce@0.0.10
 - @reggi/journey.is-promise@0.0.11
 - @reggi/journey.pass-thru@0.0.10
 - @reggi/pkg@0.0.7
 - @reggi/pkg-cli@0.0.7
 - @reggi/pkg-plugin-babel-6-to-node-4@0.0.10
 - @reggi/pkg-plugin-babel-7-to-node-4@0.0.10
 - @reggi/pkg-plugin-cobalt@0.0.10
 - @reggi/pkg-plugin-jest@0.0.10
 - @reggi/pkg-plugin-name-dir@0.0.11
 - @reggi/pkg-plugin-name-scope@0.0.11
 - @reggi/pkg-plugin-prop@0.0.11
 - @reggi/pkg-plugin-sort@0.0.11
 - @reggi/pkg-plugin-standard@0.0.10
 - @reggi/pkg.file-exists@0.0.10
 - @reggi/pkg.fs@0.0.11
 - @reggi/pkg.is-local-module@0.0.11
 - @reggi/pkg.json-parse@0.0.21
 - @reggi/pkg.pretty-json@0.0.10
 - @reggi/pkg.prop-overwrite@0.0.10
 - @reggi/pkg.read-json@0.0.9
 - @reggi/pkg.throw-error@0.0.10
 - @reggi/pkgprop@0.0.3
 - @reggi/pkgprop-cli@0.0.2
 - reggi-cli@0.0.2
 - replace-cli@1.0.10
 - @reggi/requireable@0.0.11
 - @reggi/requireable-cli@0.0.14
 - results-cli@0.0.10
 - @reggi/subrepo@0.0.3
 - @reggi/subrepo-cli@0.0.2
lerna info git Pushing tags...
lerna success publish finished
$ ssh-agent -k
unset SSH_AUTH_SOCK;
unset SSH_AGENT_PID;
echo Agent pid 25928 killed;
[ssh-agent] Stopped.
Finished: SUCCESS
```

Here's the `package.json` from my git repository.

```
{
  "name": "@reggi/pkg.json-parse",
  "version": "0.0.21",
  "description": "JSON.parse wrapped in a try / catch returns false if error",
  "author": "Thomas Reggi",
  "main": "./index.build.js",
  "scripts": {
    "babel": "babel ./index.js --out-file ./index.build.js",
    "babel:watch": "npm run babel -- --watch",
    "standard": "standard",
    "test": "jest --coverage",
    "test:result": "npm run test --silent &>/dev/null || echo $?"
  },
  "babel": {
    "plugins": [
      "transform-object-rest-spread"
    ],
    "presets": [
      [
        "env",
        {
          "targets": {
            "node": "4"
          }
        }
      ]
    ]
  },
  "jest": {
    "collectCoverage": true,
    "coverageReporters": [
      "json"
    ],
    "coverageThreshold": {
      "global": {
        "branches": 100,
        "functions": 100,
        "lines": 100,
        "statements": 100
      }
    }
  },
  "devDependencies": {
    "babel-cli": "^6.26.0",
    "babel-eslint": "^8.2.2",
    "babel-jest": "^22.4.1",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "jest": "^22.4.2",
    "standard": "^11.0.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "standard": {
    "globals": [
      "expect",
      "test",
      "beforeEach",
      "afterEach"
    ],
    "parser": "babel-eslint",
    "ignore": [
      "index.build.js"
    ]
  }
}
```

Why is it showing it is publishing but not showing the latest version in npm?

**Comment by reggi** [comment]
I have sibling packages depending on `0.0.21` and it does not exist in `npm` so those packages depending on this package are failing on install.

```
$ npm i reggi-cli -g
npm ERR! Darwin 16.7.0
npm ERR! argv "/Users/treggi/.nvm/versions/node/v6.8.0/bin/node" "/Users/treggi/.nvm/versions/node/v6.8.0/bin/npm" "i" "reggi-cli" "-g"
npm ERR! node v6.8.0
npm ERR! npm  v3.10.8
npm ERR! code ETARGET

npm ERR! notarget No compatible version found: @reggi/pkg.json-parse@^0.0.21
npm ERR! notarget Valid install targets:
npm ERR! notarget 0.0.2
npm ERR! notarget
npm ERR! notarget This is most likely not a problem with npm itself.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget
npm ERR! notarget It was specified as a dependency of '@reggi/pkg.read-json'
npm ERR! notarget

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/treggi/Desktop/acq-cps-angular-install/npm-debug.log
```

**Comment by reggi** [comment]
This might be an issue with `npm`.

> You cannot publish over the previously published versions: 0.0.21

```
➜  pkg.json-parse git:(updates) ✗ npm publish
npm notice
npm notice 📦  @reggi/pkg.json-parse@0.0.21
npm notice === Tarball Contents ===
npm notice 1.4kB package.json
npm notice 285B  index.build.js
npm notice === Tarball Details ===
npm notice name:          @reggi/pkg.json-parse
npm notice version:       0.0.21
npm notice package size:  861 B
npm notice unpacked size: 1.7 kB
npm notice shasum:        7aed9e12cdf30e3aeeb6264d7471d7e54de26849
npm notice integrity:     sha512-pSEfyJBuNXdkp[...]PEV7cWH3p/PFQ==
npm notice total files:   2
npm notice
npm ERR! publish Failed PUT 403
npm ERR! code E403
npm ERR! You cannot publish over the previously published versions: 0.0.21. : @reggi/pkg.json-parse

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/thomas/.npm/_logs/2018-05-02T16_18_09_457Z-debug.log
```

**Comment by reggi** [comment]
Also not a dist-tag issue :(

```
➜  pkg.json-parse git:(updates) ✗ npm dist-tag add @reggi/pkg.json-parse@0.0.21 latest
npm WARN dist-tag add latest is already set to version 0.0.21
```

**Comment by reggi** [comment]
I just published 0.0.22 manually without lerna.

Here's more confusion:

```
➜  pkg.json-parse git:(updates) ✗ npm publish
npm notice
npm notice 📦  @reggi/pkg.json-parse@0.0.22
npm notice === Tarball Contents ===
npm notice 1.4kB package.json
npm notice 285B  index.build.js
npm notice === Tarball Details ===
npm notice name:          @reggi/pkg.json-parse
npm notice version:       0.0.22
npm notice package size:  861 B
npm notice unpacked size: 1.7 kB
npm notice shasum:        13a3ecc553537d72adeca62b5a6ad24fd7f9231e
npm notice integrity:     sha512-9Sx+NFN6FTCDp[...]vRzQ1dUJEmf5Q==
npm notice total files:   2
npm notice
+ @reggi/pkg.json-parse@0.0.22
```

The site still has 0.0.2

https://www.npmjs.com/package/@reggi/pkg.json-parse

```
$ npm view @reggi/pkg.json-parse version
0.0.2
```

```
➜  test npm i @reggi/pkg.json-parse@latest
npm WARN test@1.0.0 No description
npm WARN test@1.0.0 No repository field.

+ @reggi/pkg.json-parse@0.0.21
updated 1 package in 1.517s
```

```
$ npm view @reggi/pkg.json-parse dist-tags
{ latest: '0.0.2' }
```

**Comment by evocateur** [comment]
In general, I don't recommend using `0.0.x` version ranges, they're very odd (and impossible to communicate breaking changes with).
